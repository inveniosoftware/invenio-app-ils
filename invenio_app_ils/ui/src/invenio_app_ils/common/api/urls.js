export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
