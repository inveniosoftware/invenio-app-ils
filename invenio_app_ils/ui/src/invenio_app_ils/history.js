import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

export const goToHandler = path => e => {
  history.push(path);
};

export const goTo = (path, state = {}) => {
  history.push(path, state);
};

export default history;
