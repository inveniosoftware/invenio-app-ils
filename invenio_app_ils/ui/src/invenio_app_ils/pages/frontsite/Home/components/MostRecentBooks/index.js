import { connect } from 'react-redux';
import { fetchMostRecentBooks } from './state/actions';
import MostRecentBooksComponent from './MostRecentBooks';

const mapStateToProps = state => ({
  data: state.mostRecentBooks.data,
  error: state.mostRecentBooks.error,
  isLoading: state.mostRecentBooks.isLoading,
  hasError: state.mostRecentBooks.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchMostRecentBooks: () => dispatch(fetchMostRecentBooks()),
});

export const MostRecentBooks = connect(
  mapStateToProps,
  mapDispatchToProps
)(MostRecentBooksComponent);
