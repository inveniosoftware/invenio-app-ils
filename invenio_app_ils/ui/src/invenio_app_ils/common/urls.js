import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
