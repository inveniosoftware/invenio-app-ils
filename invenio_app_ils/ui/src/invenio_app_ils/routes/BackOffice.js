import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { BackOfficeURLS } from '../common/urls';
import {
  Header,
  Home,
  DocumentDetailsContainer,
  DocumentsSearch,
  ItemsSearch,
  ItemDetailsContainer,
  LoansSearch,
  LoanDetailsContainer,
  LocationList,
  UserDetailsContainer,
} from '../pages/backoffice';
import './BackOffice.scss';
import { Notifications } from '../common/components/Notifications';

export class BackOffice extends Component {
  render() {
    return (
      <div>
        <Header />
        <Notifications />
        <div className="container">
          {/* home */}
          <Route exact path={BackOfficeURLS.home} component={Home} />
          {/* documents */}
          <Route
            exact
            path={BackOfficeURLS.documentDetails}
            component={DocumentDetailsContainer}
          />
          {/* documents */}
          <Route
            exact
            path={BackOfficeURLS.documentsSearch}
            component={DocumentsSearch}
          />
          {/* items */}
          <Route
            exact
            path={BackOfficeURLS.itemsSearch}
            component={ItemsSearch}
          />
          <Route
            exact
            path={BackOfficeURLS.itemDetails}
            component={ItemDetailsContainer}
          />
          {/* loans */}
          <Route
            exact
            path={BackOfficeURLS.loansSearch}
            component={LoansSearch}
          />
          <Route
            exact
            path={BackOfficeURLS.loanDetails}
            component={LoanDetailsContainer}
          />
          {/* locations */}
          <Route
            exact
            path={BackOfficeURLS.locationList}
            component={LocationList}
          />
          {/* patrons */}
          <Route
            exact
            path={BackOfficeURLS.patronDetails}
            component={UserDetailsContainer}
          />
        </div>
      </div>
    );
  }
}
