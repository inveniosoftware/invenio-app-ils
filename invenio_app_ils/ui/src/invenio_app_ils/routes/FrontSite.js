import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { FrontSiteURLS } from '../common/urls';
import { Footer, Header, Home, BookDetailsContainer } from '../pages/frontsite';
import './FrontSite.scss';
import { Notifications } from '../common/components/Notifications';

export class FrontSite extends Component {
  render() {
    return (
      <div className="app">
        <Header />
        <Notifications />
        <div className="app-content">
          {/* home */}
          <Route exact path={FrontSiteURLS.home} component={Home} />
          {/* books */}
          <Route
            exact
            path={FrontSiteURLS.bookDetails}
            component={BookDetailsContainer}
          />
        </div>
        <Footer />
      </div>
    );
  }
}
