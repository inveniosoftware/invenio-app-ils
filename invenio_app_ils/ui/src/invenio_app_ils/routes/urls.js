import { generatePath } from 'react-router-dom';
import { invenioConfig } from '../common/config';

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}records/:documentPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentsList: `${BackOfficeBase}/documents`,
  documentsCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  eitemsList: `${BackOfficeBase}/eitems`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  itemsList: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansList: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  locationsList: `${BackOfficeBase}/locations`,
  seriesList: `${BackOfficeBase}/series`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
};

const BackOfficeRouteGenerators = {
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

export const openRecordEditor = (path, recid = '') => {
  window.open(`${invenioConfig.editor.url}${path}${recid}`, '_blank');
};
