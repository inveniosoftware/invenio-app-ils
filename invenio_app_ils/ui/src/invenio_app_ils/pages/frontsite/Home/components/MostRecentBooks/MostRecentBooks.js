import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { FrontSiteRoutes } from '../../../../../routes/urls';
import { Card, Button, Divider } from 'semantic-ui-react';
import { BookCard } from '../BookCard';
import { truncate } from 'lodash/string';
import { goToHandler } from '../../../../../history';

export default class MostRecentBooks extends Component {
  constructor(props) {
    super(props);
    this.fetchMostRecentBooks = props.fetchMostRecentBooks;
    this.maxItemsToDisplay = props.maxDisplayedItems;
  }

  componentDidMount() {
    this.fetchMostRecentBooks();
  }

  prepareData(data) {
    return data.hits.slice(0, this.maxItemsToDisplay).map(document => {
      return {
        pid: document.metadata.pid,
        title: document.metadata.title,
        authors: truncate(document.metadata.authors.join('\n')),
        imageSize: 'small',
        imageCover: 'https://assets.thalia.media/img/46276899-00-00.jpg',
        eitems: document.metadata.eitems.hits,
        onClick: goToHandler(
          FrontSiteRoutes.documentDetailsFor(document.metadata.pid)
        ),
      };
    });
  }

  renderCards(books) {
    const cards = books.map(books => {
      return <BookCard key={books.pid} bookData={books} />;
    });

    return <Card.Group>{cards}</Card.Group>;
  }

  render() {
    const { data, isLoading, error } = this.props;
    const documents = this.prepareData(data);
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <div>
            <Button
              content="See All"
              onClick={goToHandler(
                FrontSiteRoutes.documentsListWithQuery(
                  '&sort=mostrecent&order=desc'
                )
              )}
            />
            <Divider hidden />
            {this.renderCards(documents)}
          </div>
        </Error>
      </Loader>
    );
  }
}

MostRecentBooks.propTypes = {
  fetchMostRecentBooks: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
};
