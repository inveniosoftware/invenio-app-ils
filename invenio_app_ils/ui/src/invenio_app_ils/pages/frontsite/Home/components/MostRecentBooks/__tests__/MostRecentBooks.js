import React from 'react';
import { shallow, mount } from 'enzyme';
import { Settings } from 'luxon';
import { FrontSiteRoutes } from '../../../../../../routes/urls';
import MostRecentBooks from '../MostRecentBooks';
import history from '../../../../../../history';

Settings.defaultZoneName = 'utc';

describe('MostRecentBooks tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  it('should load the most recent books component', () => {
    const component = shallow(
      <MostRecentBooks
        data={{ hits: [], total: 0 }}
        fetchMostRecentBooks={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch most recent books on mount', () => {
    const mockedFetchMostRecentBooks = jest.fn();
    component = mount(
      <MostRecentBooks
        data={{ hits: [], total: 0 }}
        fetchMostRecentBooks={mockedFetchMostRecentBooks}
      />
    );
    expect(mockedFetchMostRecentBooks).toHaveBeenCalled();
  });

  it('should render pending loans', () => {
    const data = {
      hits: [
        {
          metadata: {
            pid: 'doc1',
            title: 'patron_1',
            authors: [],
            _computed: { eitems: [] },
          },
        },
        {
          metadata: {
            pid: 'doc2',
            title: 'patron_2',
            authors: [],
            _computed: { eitems: [] },
          },
        },
      ],
      total: 2,
    };
    component = mount(
      <MostRecentBooks data={data} fetchMostRecentBooks={() => {}} />
    );

    expect(component).toMatchSnapshot();
    const cards = component
      .find('MostRecentBooks')
      .find('CardGroup')
      .find('BookCard')
      .find('Card')
      .filterWhere(
        element =>
          element.prop('data-test') === 'doc1' ||
          element.prop('data-test') === 'doc2'
      );
    expect(cards).toHaveLength(2);
  });

  it('should render the see all button when showing most recent books', () => {
    const data = {
      hits: [
        {
          metadata: {
            pid: 'doc1',
            title: 'patron_1',
            authors: [],
            _computed: { eitems: [] },
          },
        },
        {
          metadata: {
            pid: 'doc2',
            title: 'patron_2',
            authors: [],
            _computed: { eitems: [] },
          },
        },
      ],
      total: 2,
    };
    component = mount(
      <MostRecentBooks
        data={data}
        fetchMostRecentBooks={() => {}}
        maxItemsToDisplay={1}
      />
    );

    expect(component).toMatchSnapshot();
    const button = component
      .find('MostRecentBooks')
      .find('Button')
      .filterWhere(element => element.prop('content') === 'See All');
    expect(button).toHaveLength(1);
  });

  it('should go to book details when clicking on a book', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;
    const data = {
      hits: [
        {
          metadata: {
            pid: 'doc1',
            title: 'patron_1',
            authors: [],
            _computed: { eitems: [] },
          },
        },
        {
          metadata: {
            pid: 'doc2',
            title: 'patron_2',
            authors: [],
            _computed: { eitems: [] },
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <MostRecentBooks
        data={data}
        fetchMostRecentBooks={() => {}}
        maxItemsToDisplay={1}
      />
    );
    expect(component).toMatchSnapshot();

    const docPid = data.hits[0].metadata.pid;
    const button = component
      .find('MostRecentBooks')
      .find('CardGroup')
      .find('BookCard')
      .find('Card')
      .filterWhere(element => element.prop('data-test') === 'doc1')
      .find('a');
    button.simulate('click');

    const expectedParam = FrontSiteRoutes.documentDetailsFor(docPid);
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam);
  });
});
