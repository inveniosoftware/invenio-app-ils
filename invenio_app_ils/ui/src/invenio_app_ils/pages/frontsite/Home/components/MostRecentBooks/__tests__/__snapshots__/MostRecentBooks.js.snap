// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostRecentBooks tests should go to book details when clicking on a book 1`] = `
<MostRecentBooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc1",
            "title": Object {
              "title": "patron_1",
            },
          },
        },
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc2",
            "title": Object {
              "title": "patron_2",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentBooks={[Function]}
  maxItemsToDisplay={1}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  See All
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    See All
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                See All
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc1",
                  "title": "patron_1",
                }
              }
              key="doc1"
            >
              <Card
                data-test="doc1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_1
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc2",
                  "title": "patron_2",
                }
              }
              key="doc2"
            >
              <Card
                data-test="doc2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_2
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentBooks>
`;

exports[`MostRecentBooks tests should load the most recent books component 1`] = `
<Loader>
  <Error>
    <div>
      <Button
        as="button"
        content="See All"
        onClick={[Function]}
      />
      <Divider
        hidden={true}
      />
      <CardGroup />
    </div>
  </Error>
</Loader>
`;

exports[`MostRecentBooks tests should render recent ebooks 1`] = `
<MostRecentBooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc1",
            "title": Object {
              "title": "patron_1",
            },
          },
        },
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc2",
            "title": Object {
              "title": "patron_2",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentBooks={[Function]}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  See All
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    See All
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                See All
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc1",
                  "title": "patron_1",
                }
              }
              key="doc1"
            >
              <Card
                data-test="doc1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_1
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc2",
                  "title": "patron_2",
                }
              }
              key="doc2"
            >
              <Card
                data-test="doc2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_2
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentBooks>
`;

exports[`MostRecentBooks tests should render the see all button when showing most recent books 1`] = `
<MostRecentBooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc1",
            "title": Object {
              "title": "patron_1",
            },
          },
        },
        Object {
          "metadata": Object {
            "authors": Array [],
            "eitems": Object {
              "hits": Array [],
              "total": 0,
            },
            "pid": "doc2",
            "title": Object {
              "title": "patron_2",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentBooks={[Function]}
  maxItemsToDisplay={1}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui button"
                >
                  See All
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui button"
                  >
                    See All
                  </button>,
                }
              }
            >
              <button
                className="ui button"
                onClick={[Function]}
              >
                See All
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc1",
                  "title": "patron_1",
                }
              }
              key="doc1"
            >
              <Card
                data-test="doc1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_1
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
            <BookCard
              bookData={
                Object {
                  "authors": Array [],
                  "eitems": Array [],
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "pid": "doc2",
                  "title": "patron_2",
                }
              }
              key="doc2"
            >
              <Card
                data-test="doc2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="doc2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          patron_2
                        </div>
                      </CardHeader>
                      <CardMeta>
                        <div
                          className="meta"
                        />
                      </CardMeta>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </BookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentBooks>
`;
